music21/metadata/__init__.py

```diff
@@ -142,115 +238,376 @@ class Metadata(base.Music21Object):
+        >>> all
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Jeff Bowen>),
+        ('marcrel:LBT', <music21.metadata.primitives.Contributor librettist:Hunter Bell>),
+        ('dcterms:title', <music21.metadata.primitives.Text [title of show]>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        >>> all = md.getAllNamedValues(skipContributors=True)
+        >>> all
+        [('dcterms:title', <music21.metadata.primitives.Text [title of show]>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        '''
+        allOut: t.List[t.Tuple[str, t.Any]] = []
+
+        for nsKey, value in self._metadata.items():
+            if skipContributors and self._isContributorNSKey(nsKey):
+                continue
+
+            for v in value:
+                allOut.append((nsKey, v))
+
+        return allOut
+
+    def getAllContributorNamedValues(self) -> t.List[t.Tuple[str, t.Any]]:
+        '''
+        Returns all contributors stored in this metadata as a list of (nsKey, value) tuples.
+        The tuple's first element will be of the form 'namespace:name'.
+
+        >>> md = metadata.Metadata()
+        >>> md.addCustom('composer', 'Not a contributor')
+        >>> md.add('title', '[title of show]')
+        >>> md.addCustom('excerpt-start-measure', 1234)
+        >>> md.add('composer', ['Ludwig von Beethoven', 'Wolfgang Amadeus Mozart'])
+        >>> md.add('librettist', 'Marie Červinková-Riegrová')
+        >>> all = md.getAllContributorNamedValues()
+        >>> len(all)
+        3
+        >>> all[0]
+        ('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Ludwig von Beethoven>)
+        >>> all[1]
+        ('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Wolfgang Amadeus Mozart>)
+        >>> all[2][0]
+        'marcrel:LBT'
+        >>> all[2][1]
+        <music21.metadata.primitives.Contributor librettist:Marie Červinková-Riegrová>
+        '''
+
+        allOut: t.List[t.Tuple[str, Contributor]] = []
+
+        for nsKey, value in self._metadata.items():
+            if not self._isContributorNSKey(nsKey):
+                continue
+
+            for v in value:
+                allOut.append((nsKey, v))
+
+        return allOut
+
+# -----------------------------------------------------------------------------
+#   A few static utility routines for clients calling public APIs
+
+    @staticmethod
+    def uniqueNameToNSKey(uniqueName: str) -> t.Optional[str]:
+        '''
+        Translates a unique name to the associated standard property's NSKey.
+        Returns None if no such associated standard property can be found.
+
+        >>> metadata.Metadata.uniqueNameToNSKey('librettist')
+        'marcrel:LBT'
+        >>> metadata.Metadata.uniqueNameToNSKey('not a standard property') is None
+        True
+        >>> metadata.Metadata.uniqueNameToNSKey(None) is None
+        True
```

Comment:
this should raise a ValueError because `None` is not a string.

Reply:
ok

Reply:
Done